### ISSUE TITLE
wrong assertions on VM subscription tests

### ISSUE THREAD

**Issue by rplevka** [SatelliteQE/robottelo#4153]
e.g.:
https://github.com/SatelliteQE/robottelo/blob/master/tests/foreman/cli/test_activationkey.py#L798

It seems legit, however the fact that we don't assign any content (subscriptions) to the CV causes `subscription-manager register` to throw an error even on a successful registration:
```
 >>> subscription-manager register --org bvyeLx --activationkey AHS48doejV --force
2017-01-06 16:44:26 - robottelo.ssh - DEBUG - <<< stdout
The system has been registered with ID: e01d552c-01c0-4a5a-9c07-0921322b4e36 

Installed Product Current Status:
Product Name: Red Hat Enterprise Linux Server
Status:       Not Subscribed

Unable to find available subscriptions for all your installed products.
```

..we should evaluate such result as a success but we assert on a return code:
```
self.assertEqual(result.return_code, 0)
```

We should probably assert on a STDOUT to contain
```
The system has been registered with ID
```
or at least tolerate the 'none available subscriptions' message

**Comment by ldjebran** [comment]
Perhaps when this automation test was written the subscription-manager had a success return code but at some time the behavior of subscription-manager has changed.
Think it is very  important that subscription-manager exit code is 0.
So we have to investigate on subscription-manager if it needs an OS subscription to function normally, if so, we have to change the test to add the RHEL6 product to one subscription first.
It can happen that this is a wrong behavior of subscription manager it self.

**Comment by rplevka** [comment]
@ldjebran i think it is an overkill. The goal of the test is to find out, whether a host can be __registered__ using multiple AK's. The content||subscriptions are not an object of testing here.

**Comment by ldjebran** [comment]
@rplevka, thanks, 
Yes I agree with your statement, in this special case we have to work around the subscription issue (and perhaps log a bug for subscription manager, as the command is specific to subscribe to a content view and not to verify installed products subscription consistency).
In this special case we have to make a work around (as we don't know if this behavior will be changed or not).
By the way, we have many test that rely on command status code to verify if the command passed or not.

**Comment by rochacbruno** [comment]
@rplevka is that causing build failures?

**Comment by rplevka** [comment]
@rochacbruno it causes the tests to fail. I've got this, the fix should be easy - If you agree on what i've proposed in the description

**Comment by rochacbruno** [comment]
@rplevka agreed with your idea to assert the returned message.

**Comment by abalakh** [comment]
@rplevka 6.3 and 6.2.z should have been addressed by #4572 and #4555 accordingly.

**Comment by rplevka** [comment]
closed as this has been taken care of long time ago
